### PR TITLE
QL: fix some more style-guide violations in the alert-messages

### PR DESCRIPTION
**Pull request by erik-krogh**
One last PR, where I went through the alerts produced by QL-for-QL.

And I removed some benign results in e.g. meta queries.

### CODE REVIEW

File: ql/ql/src/queries/performance/TransitiveStep.ql [hunk 1]
```diff
@@ -161,5 +161,5 @@ predicate transitiveDelta(Call rec, TransitivelyClosedCall tc) {
 
 from Call rec, TransitivelyClosedCall tc, AstNode reason
 where transitiveDelta(rec, tc) and reason = tc.getReason()
-select tc, "This recursive delta is transively closed $@, which may be a performance problem.",
-  reason, "here"
+select tc, "This recursive delta $@, which may be a performance problem.", reason,
+  "is transively closed"
```
Comment (tausbn) [inline ql/ql/src/queries/performance/TransitiveStep.ql:165]:
```suggestion
select tc, "This recursive delta is $@, which may be a performance problem.", reason,
  "transitively closed"
```
Typo (and moved the "is" outside of the link).

File: ql/ql/src/queries/performance/UnusedField.ql [hunk 2]
```diff
@@ -31,5 +31,5 @@ where
     )
   ) and
   (if clz = implClz then extraMsg = "." else extraMsg = " of any class between it and $@.")
-select clz, "The field $@ declared in $@ is not used in the characteristic predicate" + extraMsg,
-  field, field.getName(), clz, clz.getName(), implClz, implClz.getName()
+select clz, "This class declares $@ that is not used in the characteristic predicate" + extraMsg,
+  field, "field " + field.getName(), implClz, implClz.getName()
```
Comment (tausbn) [inline ql/ql/src/queries/performance/UnusedField.ql:35]:
```suggestion
select clz, "This class declares the $@ but does not bind it in the characteristic predicate" + extraMsg,
  field, "field " + field.getName(), implClz, implClz.getName()
```
Reformulated to read a bit better (to me at least).

File: ql/ql/src/queries/style/AlertMessage.ql [hunk 3]
```diff
@@ -126,7 +126,7 @@ String avoidHere(Select sel, string part) {
 String avoidArticleInLinkText(Select sel) {
   result = sel.getExpr((any(int i | i > 1))) and
   result = getSelectPart(sel, _) and
-  result.getValue().regexpMatch("a|an .*")
+  result.getValue().regexpMatch("(a|an) .*")
```
Comment (tausbn) [inline ql/ql/src/queries/style/AlertMessage.ql:129]: 😱

File: ql/ql/src/queries/style/DeadCode.ql [hunk 4]
```diff
@@ -12,4 +12,4 @@ import codeql_ql.style.DeadCodeQuery
 
 from AstNode node
 where isDead(node)
-select node, "Code is dead"
+select node, "Code is dead."
```
Comment (tausbn) [inline ql/ql/src/queries/style/DeadCode.ql:15]: While I appreciate the nihilism expressed by this statement, perhaps "This code is never used." would be more in line with our other alert messages.

**Review by tausbn — CHANGES_REQUESTED** [review]
A few comments here and there, but otherwise this looks good to me. 👍

File: ql/ql/src/queries/style/AlertMessage.ql (hunk 3 above)
Comment (erik-krogh) [reply ql/ql/src/queries/style/AlertMessage.ql:129]: Agreed.

**Review by tausbn — APPROVED** [review]

**Pull request merged**